R/which_element.R

```diff
@@ -0,0 +1,42 @@
+#' which_element
+#'
+#' This function returns locations within an EML list object that contain an instance where \code{element == value}
+#' @param eml_list an EML list object
+#' @param element element to evaluate
+#' @param value string to match \code{element} to
+#'
+#' @keywords eml
+#'
+#' @examples
+#' which_element(eml@@dataset@@creator,"surName","Smith")
+#'
+#' which_element(eml@@dataset@@dataTable,"objectName","file.csv")
+#'
+#' which_element(eml@@dataset@@dataTable[[1]]@@attributeList@@attribute,"numberType","real")
+#' @export
+which_element <- function(eml_list, element, value) {
+
+    stopifnot(isS4(eml_list))
+    stopifnot(is(eml_list,"list"))
+    stopifnot(is.character(element))
+    stopifnot(is.character(value))
+
+    # find location
+    location <- unlist(lapply(seq_along(eml_list), function(i) { # eml_get returns NULL if field not present so need to loop
```

Comment:
do you want to print a message for when `seq_along = 0`?  Like list not populated or something similar